doc/contributing/adding-v8-fast-api.md

```diff
@@ -1,91 +1,396 @@
+
+```cpp
+bool FastIsObject(v8::Local<v8::Value>, // receiver, discarded
+                  v8::Local<v8::Value> value) {
+  return value->IsObject();
+}
+```
+
+### Appending an `options` argument
+
+Fast callbacks may add an optional final function argument of type
+`const v8::FastApiCallbackOptions&`. This is required if the callback interacts
+with the isolate in any way: see
+[Stack-allocated objects and garbage collection](#stack-allocated-objects-and-garbage-collection)
+and [Handling errors](#handling-errors).
+
+```cpp
+void FastThrowExample(const int32_t n,
+                      const v8::FastApiCallbackOptions& options) {
+  if (IsEvilNumber(n)) {
+    v8::HandleScope handle_scope(options.isolate);
+    THROW_ERR_INVALID_ARG_VALUE(options.isolate, "Begone, foul spirit!");
+  }
+}
+```
+
+## Registering a Fast API callback
+
+Compare registering a conventional API binding:
+
+```cpp
+void Initialize(Local<Object> target,
+                Local<Value> unused,
+                Local<Context> context,
+                void* priv) {
+  Environment* env = Environment::GetCurrent(context);
+  SetMethodNoSideEffect(context, target, "isEven", IsEven);
+}
+```
+
+with registering an API binding with a fast callback:
+
+```cpp
+void Initialize(Local<Object> target,
+                Local<Value> unused,
+                Local<Context> context,
+                void* priv) {
+  Environment* env = Environment::GetCurrent(context);
+  SetFastMethodNoSideEffect(context,
+                            target,
+                            "isEven",
+                            SlowIsEven,
+                            &fast_is_even);
+}
+```
+
+The Fast API equivalents of the method binding functions take an additional
+parameter, which specifies the fast callback(s).
+
+In the majority of cases, there will only be a single fast callback, and the
+additional parameter should be a pointer to the `v8::CFunction` object
+constructed by the call to `CFunction::Make`.
+
+In rare cases, there may be more than one fast callback, _eg._ if the function
+accepts optional arguments. In this case, the additional parameter should be a
+reference to an array of `v8::CFunction` objects, which is used to initialize a
+`v8::MemorySpan<v8::CFunction>`:
+
+```cpp
+uint32_t FastFuncWithoutArg() {
+  return -1;
+}
+uint32_t FastFuncWithArg(const v8::FastOneByteString& s) {
+  return s.length;
+}
+static CFunction fast_func_callbacks[] = {CFunction::Make(FastFuncWithoutArg),
+                                          CFunction::Make(FastFuncWithArg)};
+
+void Initialize(Local<Object> target,
+                Local<Value> unused,
+                Local<Context> context,
+                void* priv) {
+  Environment* env = Environment::GetCurrent(context);
+  SetFastMethodNoSideEffect(context,
+                            target,
+                            "func",
+                            SlowFunc,
+                            fast_func_callbacks);
```

Comment:
Passing the array directly (corresponding to the array signature of the `v8::MemorySpan` constructor) seems like the more straightforward recommendation. The only example in the codebase at present uses the iterator signature:

https://github.com/nodejs/node/blob/0992d28011effcd1922342a82e879cb22184e0fc/src/node_url.cc#L525-L526